compass/hierarchy.py

```diff
@@ -150,9 +152,15 @@ def get_hierarchy(self, compass_id: int, level: str) -> dict:
         except FileNotFoundError:
             pass
 
+        # Fetch the hierarchy
         out = self._get_descendants_recursive(compass_id, hier_level=level)
-        with open(filename, "w", encoding="utf-8") as f:
-            json.dump(out, f, ensure_ascii=False)
+
+        # Try and write to a file for caching
+        try:
+            with open(filename, "w", encoding="utf-8") as f:
+                json.dump(out, f, ensure_ascii=False)
+        except IOError as x:
+            print(f"Unable to write cache file: {x.errno} - {x.strerror}")
```

Comment:
I assume these are for convenience (I ususally have write permissions locally)! 

Perhaps add a helper function in utility.py as the try/except code is exactly the same?

```suggestion
        except IOError as e:
            print(f"Unable to write cache file: {e.errno} - {e.strerror}")
```